hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/TransactionModule.java

```diff
@@ -53,6 +55,12 @@
 
 @Module(includes = {TransactionConfigModule.class, TransactionInitialStateModule.class})
 public interface TransactionModule {
+    @Provides
+    @TransactionScope
+    static CustomGasCalculator provideCustomGasCalculator(@NonNull HederaFunctionality functionality) {
+        return new CustomGasCalculator();
```

Comment:
@tinker-michaelj 

Would you take a look at the latest commit? I've added `CustomGasCalculator` to `Transaction` and `Query` scopes. I was thinking about how to inject `Query/Handle` context or `HederaFunctionality functionality` from Component to `new CustomGasCalculator()`. I have no success with passing `functionality` to the `CustomGasCalculator` constructor :confused:. 

Do you have something in mind about how to use `context` or `functionality` in `CustomGasCalculatorl.logOperationGasCost` method?

Reply:
Hi @natanasow, do you mind if I open a PR to this branch? 

Or maybe we need an out-of-band call; some of these details are definitely hard to nail down asynchronously via GitHub. 😇

Reply:
@tinker-michaelj feel free to open a completely new branch and or commit whatever you want to this one.